pkg/stanza/fileconsumer/file_sort.go

```diff
@@ -104,30 +104,21 @@ type TimestampSortRule struct {
 	Location     string `mapstructure:"location,omitempty"`
 }
 
-func (f TimestampSortRule) validate() error {
+func (f *TimestampSortRule) validate() error {
 	if f.RegexKey == "" {
 		return fmt.Errorf("regex key must be specified for timestamp sort")
 	}
 	if f.Layout == "" {
 		return fmt.Errorf("format must be specified for timestamp sort")
 	}
+
 	if f.Location == "" {
 		f.Location = "UTC"
 	}
-
-	loc, err := timeutils.GetLocation(&f.Location, nil)
-	if err != nil {
-		return fmt.Errorf("parse location %s: %w", f.Location, err)
-	}
-
-	_, err = timeutils.ParseStrptime(f.Layout, "", loc)
-	if err != nil {
-		return fmt.Errorf("parse format %s: %w", f.Layout, err)
-	}
 	return nil
```

Comment:
I don't think removing validation altogether is the right approach here. Please validate the layout [as is done in the time parser](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d5d2bc04752a289aa9712fbc622adc115c9be71d/pkg/stanza/operator/helper/time.go#L83-L86).

Reply:
Updated